partiql-test-data/partiql_test_schema.isl

```diff
@@ -0,0 +1,103 @@
+schema_header::{
+}
+
+type::{
+    name: PartiQLTestDocument,
+    type: document,
+    element: { one_of: [TestCase, Namespace] }
+}
+
+type::{
+    name: Namespace,
+    type: list,
+    element: { one_of: [TestCase, Namespace] }
+}
+
+type::{
+    name: TestCase,
+    type: { one_of: [PassParseTestCase, FailParseTestCase, FailSemanticTestCase] }
+}
```

Comment:
Does it make sense to have a more abstract model for `TestCase`, and then re-use the model for `PassParseTestCase`, etc., e.g.:

```
type::{
    name: TestCase,
    type: struct,
    annotations: [ parse, semantic ], // Not sure if/how we can express: at most one of these annotations
    fields: {
        name: { type: string, occurs: required },
        statement: { type: string, occurs: required },
        assert: {
            type: {
                one_of: [
                    ParseOkAssertion,
                    { type: list, contains: [{result: ParseOk}] },
                    // ...
                ]
            },
            occurs: required
        }
    }
}
```

BTW, I'm still learning `ISL` so please take the above with the grain of salt.

Reply:
Agree. Perhaps another way to think about / model this:

Every test case has:
 - 1 name
 - 1 or more statements
 - 1 or more assertions

If we do this, there's actually no need to have the `parse`|`semantic` annotation - it's the assertion that's important. It seems redundant to have the annotation and the assertions

So:
- A syntax test might have: 1 name, 1 statement, 1 `SyntaxSuccess` assertion.

- A static analysis check test might have: 1 name, 1 statement, 1 `StaticAnalysisSuccess` assertion

- A static analysis *failure* check test might have: 1 name, 1 statement, 1 `SyntaxSuccess` assertion and 1 `StaticAnalysisFail` assertion

- An evaluator test might have: 1 name, 1 statement, 1 `EvaluatorSuccess` assertion with embedded data.

But we could also have something like an assertion that multiple queries are equivalent (e.g. `SELECT 10` and `SELECT 5*2` and `SELECT 20/2` should allreturn the same result) that would have: 1 name, multiple statements,  1 `EvaluatorSuccess` assertion with embedded data.


Of course, doing the above would require reworking the existing test data, but if it's the right data model, better to do now than when we have even more test cases.

Reply:
Second that, I think we've had couple discussions in the previous PRs regarding if we need annotations at all and the more we move forward the more I see we don't.

Reply:
> If we do this, there's actually no need to have the `parse|semantic` annotation - it's the assertion that's important. It seems redundant to have the annotation and the assertions

Also agree. The more that I look at the defined test data the less value I see in having both the annotation and the assertion. The annotation does make it slightly easier for the test runner to parse the test case. But considering the redundancy for every test, I think it's worth eliminating the annotation at this time.

Regarding the other test categories, I'll see how I can best model those other tests and how they can be modeled in ISL.

Reply:
Regarding the equivalence tests, added an issue to track https://github.com/partiql/partiql-tests/issues/17